pkg/cmd/cli/cmd/newapp.go

```diff
@@ -90,6 +90,19 @@ application is created.`
 
   // Search for "ruby" in stored templates and print the output as an YAML
   $ %[1]s new-app --search --template=ruby --output=yaml`
+
+	newAppNoInput = `You must specify one or more images, image streams, templates or source code locations to create an application.
+
+Suggestions:
+Use '-L' or '--list' flag to list all local templates and image streams that can be used to create an application
```

Comment:
Also, it would be better to compress this help message down - it's fairly large, and it would be ok to describe list and search.  You only need one example for search ("you can search with oc new-app -S, for instance oc new-app -S php").